dashboard/app/views/layouts/_header.html.haml

```diff
@@ -102,26 +100,55 @@
 - if ["/projects", "/projects/public", "/projects/libraries"].include? request.path
   #header-banner.header-banner-short{style: 'background-image: url("/shared/images/banners/project-banner.jpg")'}
 
-- if should_show_progress
-  - view_as = @user || current_user
-  - user_progress = current_user == nil ? 'null' : summarize_user_progress(script, view_as, view_as == current_user && @public_caching).to_json
+:ruby
```

Comment:
:+1: To making this a ruby block instead of lots of individual ruby lines!  I'd suggest going a step further and pulling this out of the HAML file entirely, if possible.

Reply:
I'd love to do that, but might save it up for a followup PR, because I'm running out of steam on this one :)